fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java

```diff
@@ -453,15 +456,16 @@ protected void runWaitingTxnJob() throws AlterCancelException {
                     long shadowIdxId = entry.getKey();
                     MaterializedIndex shadowIdx = entry.getValue();
                     for (Tablet shadowTablet : shadowIdx.getTablets()) {
-                        Long backendId = Utils.chooseNodeId((LakeTablet) shadowTablet);
-                        if (backendId == null) {
+                        ComputeNode computeNode = GlobalStateMgr.getCurrentState().getWarehouseMgr()
+                                .getComputeNodeAssignedToTablet(WarehouseManager.DEFAULT_WAREHOUSE_ID, (LakeTablet) shadowTablet);
```

Comment:
‌‌‌‌‌Before, the compute Node Id was directly obtained through `Utils.chooseNodeId(lakeTablet)`. Why switch to this method? Are there any other considerations?

Reply:
Utils itself is a poorly designed class. The management of nodes should be gathered into WarehouseManager.